frb_dart/lib/src/cli/build_web/executor.dart

```diff
@@ -149,8 +153,12 @@ Future<String> _getRustCreateName({required String rustCrateDir}) async {
   return rustCrateName;
 }
 
+Future<int> _getRustContentHash({required String rustCrateDir}) async {
+  throw UnimplementedError();
```

Comment:
@fzyzcjy how would you fetch the rust content hash from the build-web command?

Reply:
One way may be parsing the generated code and find it.

But I am thinking about something more general: How does normal wasm-pack + js users (i.e. *not* wasm + dart users) handle this? I guess there may be some ways to add content hashing within wasm-pack ecosystem, and maybe we can utilize it.

Reply:
Maybe we can just build it like it was, and then at the final packing we hash the wasm file, append it to the filename and put it in the loader config too?

Reply:
Looks reasonable! More brainstorms based on that:

* Instead of `rust_lib-abcdef.js`, we do `rust_lib.js?dummy_query=abcdef` (where `abcdef` is the content hash). Then we do not need to rename anything to keep it simple.
* Then, after `build-web`, if this feature is enabled, we just compute `hash(read_file(rust_lib.js) + read_file(rust_lib.wasm))`, and write it to a field in `frb_generated.dart`.

I am still wondering whether there are more "native" solutions instead of hand crafting it...

Reply:
You’re right, we could do that. Unfortunately the generated js file split the filename.js and add _bg.wasm as suffix. We would need to do a find and replace to add _bg.wasm?dummy=abcdef

so maybe having it in the directly would be better / easier.

Reply:
So, show we ask wasm-pack if they want to handle the hash on their side, or should I continue to work on a implementation here?

Reply:
I guess we can create an issue there to have a discussion with them (feel free to cc me if you like). Because if this feature is done there, then not only frb but also every other package that depend on wasm-pack will benefit from this!